Source/control.cpp

```diff
@@ -493,7 +493,7 @@ spell_type GetSBookTrans(spell_id ii, bool townok)
 		st = RSPLTYPE_SKILL;
 	}
 	if (st == RSPLTYPE_SPELL) {
-		if (!CheckSpell(MyPlayerId, ii, st, true)) {
+		if (SpellFailReason::Success != CheckSpell(MyPlayerId, ii, st, true)) {

```

Comment:
This will always be true as None is always returned where Success could have. Pick one :)

Reply:
`none` and `success` are synonyms, so its not always true. But I guess your are right - only one should be used.
Any naming suggestions? 

@julealgon suggested changing the result enum to  `SpellCheckingResult`  
values:
- Success
- Fail_Level0
- Fail_NoMana
- Fail_Busy (does this ever happen? If yes - how to achieve this case?)

This maybe could also used for staff and scroll cast tests.
For this 
- Fail_NoCharge
- Fail_NoScroll
- Fail_WronPlace

Reply:
`FailReason == Success` doesn't make sens to me, it seams a different category.

`Result == Success/Fail` seams fine.